src/OpenTelemetry/Metrics/MetricReaderExt.cs

```diff
@@ -157,6 +161,13 @@ internal List<Metric> AddMetricsListWithViews(Instrument instrument, List<Metric
                     else
                     {
                         Metric metric = new(metricStreamIdentity, this.GetAggregationTemporality(metricStreamIdentity.InstrumentType), this.maxMetricPointsPerMetricStream, this.exemplarFilter);
+                        if (this.parentProvider?.GetSdkHealthReporter() != null)
```

Comment:
Is this check needed?

Reply:
Yes - as we are planning to add this feature as experimental, we would only add it if the `SdkHealthReporter` is initialized for the provider.

Reply:
Add this here as well then? https://github.com/open-telemetry/opentelemetry-dotnet/pull/4442/files#diff-8650a8083e99ad454eac9d5bef1a6061cc0c1ab31085a1d25e5c4f57ed16f74fR77

Reply:
Oh, I missed it there. Will add it